src/components/fields/Percentage/Settings.tsx

```diff
@@ -0,0 +1,186 @@
+import { useEffect, useState } from "react";
+
+import {
+  Box,
+  Checkbox,
+  Grid,
+  InputLabel,
+  MenuItem,
+  TextField,
+  Typography,
+  useTheme,
+} from "@mui/material";
+import ColorPickerInput from "@src/components/ColorPickerInput";
+import { ISettingsProps } from "@src/components/fields/types";
+
+import { Color, toColor } from "react-color-palette";
+import { fieldSx } from "@src/components/SideDrawer/utils";
+import { resultColorsScale, defaultColors } from "@src/utils/color";
+import { GlobalStyles } from "tss-react";
+
+const colorLabels: { [key: string]: string } = {
+  0: "Start",
+  1: "Middle",
+  2: "End",
+};
+
+export default function Settings({ onChange, config }: ISettingsProps) {
+  const [colorsDraft, setColorsDraft] = useState<any[]>(
+    config.colors ? config.colors : defaultColors
+  );
+
+  const [checkStates, setCheckStates] = useState<boolean[]>([
+    colorsDraft[0],
+    colorsDraft[1],
+    colorsDraft[2],
+  ]);
+
+  useEffect(() => {
+    onChange("colors")(colorsDraft);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [colorsDraft]);
+
+  const onCheckboxChange = (index: number, checked: boolean) => {
+    setColorsDraft(
+      colorsDraft.map((value: any, idx: number) =>
+        index === idx ? (checked ? value || defaultColors[idx] : null) : value
+      )
+    );
+    setCheckStates(
+      checkStates.map((value, idx) => (index === idx ? checked : value))
+    );
+  };
+
+  const handleColorChange = (index: number, color: Color): void => {
+    setColorsDraft(
+      colorsDraft.map((value, idx) => (index === idx ? color.hex : value))
+    );
+  };
+
+  return (
+    <>
+      <GlobalStyles
+        styles={{
+          "& .MuiList-root.MuiMenu-list": {
+            padding: 0,
+          },
+        }}
+      />
```

Comment:
This shouldn’t be a part of GlobalStyles. It may affect other dropdowns. You can include it in the `<TextField>` `sx` prop below.

```suggestion
```